File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/request/RequestHandshake.java
package com.armongate.mobilepasssdk.model.request;

/**
 * Request model for handshake endpoint
 * Used to establish ephemeral keys for secure request signing
 */
public class RequestHandshake {
    /**
     * Member ID identifying the user
     */
    public String memberId;

    /**
     * Current timestamp in seconds (Unix epoch)
     */
    public long timestamp;

    /**
     * Client-generated ephemeral key (32 bytes, base64-encoded)
     */
    public String ephemeralKey;

    /**
     * HMAC-SHA256 signature: HMAC(STATIC_KEY, memberId + timestamp + ephemeralKey)
     */
    public String signature;
}

File: ios/MobilePassSDK/Model/Request/RequestHandshake.swift
//
//  RequestHandshake.swift
//  MobilePassSDK
//
//  Request model for handshake (ephemeral key registration)
//

import Foundation

struct RequestHandshake: Codable {
    /// Member ID from configuration
    let memberId: String
    
    /// Current timestamp (Unix seconds)
    let timestamp: Int64
    
    /// Ephemeral key (base64 encoded, 32 bytes)
    let ephemeralKey: String
    
    /// HMAC signature of the handshake payload
    let signature: String
    
    var dictionary: [String: Any] {
        return [
            "memberId": memberId,
            "timestamp": timestamp,
            "ephemeralKey": ephemeralKey,
            "signature": signature
        ]
    }
}
